### PR TITLE
Feature/mx 1581 refine publisher

### PR DESCRIPTION
**Pull request by esinsj**
# Changes
- refine pubilishing pipeline

### CODE REVIEW

File: mex/extractors/publisher/extract.py [hunk 1]
```diff
@@ -1,28 +1,25 @@
...
-    item_counter = 0
+    item_number_limit = 100  # 100 is the maximum possible number per get-request
 
-    while item_counter <= total_item_number:
-        response = connector.request(
-            method="GET",
-            endpoint="merged-item",
-            params={"limit": str(item_number_limit), "skip": str(item_counter)},
+    logging_counter = 0
+    for item_counter in range(0, total_item_number + 1, item_number_limit):
```
Comment (cutoffthetop) [inline mex/extractors/publisher/extract.py:18]: do we need this `+ 1`? e.g. when i have exactly 800 items, i could get those in 8 requests, not 9 - right?

```python
>>> total_item_number = 800
>>> item_number_limit = 100
>>> list(range(0, total_item_number + 1, item_number_limit))
[0, 100, 200, 300, 400, 500, 600, 700, 800]
```

File: mex/extractors/logging.py [hunk 2]
```diff
@@ -22,3 +22,24 @@ def log_filter(
...
+
+
+def log_processed_merged_items(
+    action: str,
+    logging_counter: int,
+    total: int | None = None,
+) -> None:
+    """Log the handling of merged items during publishing."""
+    if total:
+        logger.info(
```
Comment (cutoffthetop) [inline mex/extractors/logging.py:34]: i'd move these logger.info statements right into the code, this function doesn't help much with readability imho

File: mex/extractors/publisher/filter.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+
+def filter_merged_items(
+    items: Iterable[MergedItem],
+) -> Generator[MergedItem, None, None]:
+    """Filter to be published items by allow list."""
+    settings = Settings.get()
+
+    logging_counter = 0
+    total_counter = 0
```
Comment (cutoffthetop) [inline mex/extractors/publisher/filter.py:15]:
```suggestion
    skipped_items = 0
    total_items = 0
```
maybe that'd be a bit clearer?

File: mex/extractors/publisher/main.py [hunk 4]
```diff
@@ -1,16 +1,19 @@
...
 @asset(group_name="publisher")
 def publish_merged_items() -> None:
-    """Get all merged items from mex-backend and write to ndjson file."""
+    """Get merged items from mex-backend, filter, and write to ndjson file."""
     items = get_merged_items()
 
-    write_merged_items(items)
+    allowed_items = filter_merged_items(items)
+
+    write_merged_items(allowed_items)
```
Comment (cutoffthetop) [inline mex/extractors/publisher/main.py:16]: what do you think of splitting the get+filter and write into 2 separate assets?

**Review by cutoffthetop — CHANGES_REQUESTED** [review]
nice and clean overall! just a few simplications/suggestions from my side

File: mex/extractors/publisher/extract.py (hunk 1 above)
Comment (esinsj) [reply mex/extractors/publisher/extract.py:18]: we do not only not need it, it even throws an error. Thanks for pointing that out.

File: mex/extractors/logging.py (hunk 2 above)
Comment (esinsj) [reply mex/extractors/logging.py:34]: Do you mean to ditch the log_processed_merged_items function and write the logging directly into the extract, filter, and the write functions?

File: mex/extractors/publisher/main.py (hunk 4 above)
Comment (esinsj) [reply mex/extractors/publisher/main.py:16]: I could do that. Can you explain what the purpose is? Is it simply best practice, even though the write asset would just call the rite_merged_items function?

Comment (esinsj) [reply mex/extractors/publisher/main.py]: ah 💡 thanks!

File: mex/extractors/logging.py (hunk 2 above)
Comment (cutoffthetop) [reply mex/extractors/logging.py]: that was my suggestion yeah. but this is rather a matter of taste, if you think it makes the logging easier to read, go ahead and keep it

Comment (cutoffthetop) [reply mex/extractors/logging.py]: ah, too late ([9ac94b5](https://github.com/robert-koch-institut/mex-extractors/pull/262/commits/9ac94b5e8f3a4c0444e8222af89f19d9d26aca25)), just as well

**Review by cutoffthetop — APPROVED** [review]
looks good! let's ship 🚢

**Pull request merged**